utilities/generate_messages_catalog.py

```diff
@@ -291,6 +281,30 @@ def _build_id_messages(python_module):
     return id_messages
 
 
+def _get_message_codes(msg_code_arg):
+    """
+    Get the correct message codes based on instance type of msgCodeArg
+
+    :param msg_code_arg: the current arg to pull msgCodes from
+    :type msg_code_arg: :class:`~ast.AST`
+    :return: the correct message code to use
+    :rtype: str
+    """
+    msg_codes = None
+    if isinstance(msg_code_arg, ast.Str):
+        msg_codes = (msg_code_arg.s,)
+    else:
+        if any(isinstance(element, (ast.Call, ast.Name))
+               for element in ast.walk(msg_code_arg)):
+            msg_codes = ('(dynamic)',)
+        else:
+            msg_codes = [
+                element.s for element in ast.walk(msg_code_arg)
+                if isinstance(element, ast.Str)
+            ]
```

Comment:
Not really. Strings have a bunch of extra stuff added to them, but moreso, if you were to iterate over this string you would get character by character, not string by string.